File: solutions/labs/lab16/EncryptionView.java
import java.util.*;

public class EncryptionView {
    private Scanner s;

    public EncryptionView() {
        s = new Scanner(System.in);
    }

    public String getMessage() {
        System.out.print("Enter message: ");
        return s.nextLine();
    }

    public int getKey() {
        System.out.print("Enter the key: ");
        return Integer.parseInt(s.nextLine());
    }

    public void printError() {
        System.out.println("Error: key should be an integer.");
    }

    public void printResult(String msg, String type) {
        System.out.printf("Result from %s: %s%n", type, msg);
    }
}

File: solutions/labs/lab23/EncryptionView.cs
using System;

public class EncryptionView {
    public string GetMessage() {
        Console.Write("Enter message: ");
        string? msg = Console.ReadLine();
        return msg == null ? "" : msg;
    }

    public int GetKey() {
        Console.Write("Enter the key: ");
        return Convert.ToInt32(Console.ReadLine());
    }

    public void PrintError() {
        Console.WriteLine("Error: key should be an integer.");
    }

    public void PrintResult(string msg, string type) {
        Console.WriteLine($"Result from {type}: {msg}");
    }
}